app/controllers/devise_sessions_controller.rb

```diff
@@ -2,55 +2,80 @@ class DeviseSessionsController < Devise::SessionsController
   include ApplicationHelper
   include CookiesHelper
 
+  skip_before_action :verify_authenticity_token, only: [:create]
```

Comment:
So... having asked for this, I now realise it's not necessary.

With our session cookie having `SameSite=lax`, cookies don't get sent on cross-domain POST requests.  So clicking the big green button on the save your results page will always log the user out.  Previously we'd then have decided whether to give a login or registration journey based on email address---but we're not doing that in this PR.

So the big green button will have to go to the registration page (or the welcome page, which will redirect the user to the registration page because they won't have a session cookie), not this page.  Meaning we don't need this `skip_before_action`.

This has caused me to realise some things about CSRF in the registration controller, so I'll add a comment there too.

Reply:
We're going to keep the cross-site stuff in the welcome controller.

In which case this controller doesn't need `skip_before_action` or `params[:jwt]`.

Reply:
I'll remove the skip_before_action and use of params[:jwt] from the login and registration controllers, keeping them only in the welcome controller.